call-home/src/transmitter/client.rs

```diff
@@ -0,0 +1,51 @@
+use crate::common::{constants::*, errors::ReceiverError};
+use chrono::prelude::*;
+use k8s_openapi::chrono;
+use reqwest::Response;
+use reqwest_middleware::{ClientBuilder, ClientWithMiddleware};
+use reqwest_retry::{policies::ExponentialBackoff, RetryTransientMiddleware};
+
+/// Struct used to make calls to the receiver API.
+pub struct Receiver {
+    cluster_id: String,
+    client: ClientWithMiddleware,
+    url: String,
+}
+
+impl Receiver {
+    /// 'Receiver::new()' creates a new instance of Receiver
+    /// which is initialized with sane default values.
+    pub(crate) async fn new<T>(cluster_id: T) -> Result<Self, ReceiverError>
+    where
+        T: ToString,
+    {
+        // Retry up to 3 times with increasing intervals between attempts.
+        let retry_policy = ExponentialBackoff::builder().build_with_max_retries(3);
+
+        let client_config = reqwest::Client::builder()
+            .danger_accept_invalid_certs(true)
+            .build()?;
+        Ok(Self {
+            cluster_id: cluster_id.to_string(),
+            client: ClientBuilder::new(client_config)
+                .with(RetryTransientMiddleware::new_with_policy(retry_policy))
+                .build(),
+            url: RECEIVER_API_ENDPOINT.to_string(),
+        })
+    }
+
+    /// 'post()' method attempts an HTTP POST with some headers
+    pub(crate) async fn post(&self, body: Vec<u8>) -> reqwest_middleware::Result<Response> {
```

Comment:
It's a bit confusing having a module exposing two functions with two seemingly completely different error types